pkg/api/steve/clusters/kubeconfig.go

```diff
@@ -0,0 +1,78 @@
+package clusters
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/rancher/apiserver/pkg/types"
+	"github.com/rancher/rancher/pkg/features"
+	"github.com/rancher/rancher/pkg/kubeconfig"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/rancher/pkg/user"
+	"github.com/rancher/wrangler/pkg/schemas/validation"
+	"k8s.io/apiserver/pkg/endpoints/request"
+)
+
+type kubeconfigDownload struct {
+	userMgr user.Manager
+}
+
+func (k kubeconfigDownload) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	apiRequest := types.GetAPIContext(req.Context())
+	if err := apiRequest.AccessControl.CanGet(apiRequest, apiRequest.Schema); err != nil {
+		apiRequest.WriteError(err)
+		return
+	}
+
+	if features.MCM.Enabled() {
+		http.Redirect(rw, req, fmt.Sprintf("/v3/clusters/%s?action=generateKubeconfig", apiRequest.Name), http.StatusFound)
```

Comment:
Is this required, can we handle the token here anyways?

Reply:
Not really, the old logic handles something like localAuthEndpoint which is no longer avaiable if you run MCM=false, so I would rather redirect the response if MCM is enabled.

Reply:
Makes sense